$OpenBSD$

Disable TCP keepalive tuning on OpenBSD.
OpenBSD kernel does not support TCP_KEEPIDLE/TCP_KEEPINTVL/TCP_KEEPCNT
socket options. The constants exist in <netinet/tcp.h> but setsockopt
returns ENOPROTOOPT for all of them. Make setSocketKeepAlive a no-op.

--- src/Simplex/Messaging/Transport/KeepAlive.hs.orig
+++ src/Simplex/Messaging/Transport/KeepAlive.hs
@@ -54,8 +54,13 @@
 #endif

+#if defined(openbsd_HOST_OS)
+setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
+setSocketKeepAlive _ _ = pure ()
+#else
 setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
 setSocketKeepAlive sock KeepAliveOpts {keepCnt, keepIdle, keepIntvl} = do
   setSocketOption sock KeepAlive 1
   setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPIDLE) keepIdle
   setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPINTVL) keepIntvl
   setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPCNT) keepCnt
+#endif

 $(J.deriveJSON defaultJSON ''KeepAliveOpts)
